packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1230,6 +1230,47 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     });
   }
 
+  /// Whether this has been disposed.
+  ///
+  /// If asserts are disabled, this property is always null.
+  bool? get debugDisposed {
+    bool? disposed;
+    assert(() {
+      disposed = _debugDisposed;
+      return true;
+    }());
+    return disposed;
+  }
+
+  bool _debugDisposed = false;
+
+  /// Release any resources held by this render object.
+  ///
+  /// If this render object has created any children directly, it should dispose
+  /// of those children in this method as well. However, it should not
+  /// dispose of children that were created by some other object, such as
+  /// a [RenderObjectElement].
+  ///
+  /// If [isRepaintBoundary] returns true, the layer tree rooted at this
+  /// object's layer will also dispose.
+  @mustCallSuper
+  void dispose() {
+    assert(!_debugDisposed);
+    if (isRepaintBoundary) {
+      _layer = null;
+    }
```

Comment:
Why only null this out if we are a repaint boundary? Shouldn't we release all resources?

Reply:
What other resources does the top level `RenderObject` have?

Children will get dropped or disposed by elements holding them, unless we ourselves created the children, which I'll add to the top level documentation on RO.

Reply:
What I meant is: "Why only null our `_layer` if it is a repaint boundary? If you're not a repaint boundary, who is responsible for freeing up this resource?

Reply:
Ahhh, I guess we could null out our reference. In another branch, this code also calls `dispose` on thelayer, which we should only do if we're a repaint boundary. But yeah we can unconditionally null it out.